MSALiOSB2C/ViewController.swift

```diff
@@ -363,16 +338,33 @@ class ViewController: UIViewController, UITextFieldDelegate, URLSessionDelegate
         }
     }
     
-    func getUserByPolicy (withUsers: [MSALUser], forPolicy: String) throws -> MSALUser? {
+    func getAccountByPolicy (withAccounts: [MSALAccount], forPolicy: String) throws -> MSALAccount? {
         
-        for user in withUsers {
-            if (user.userIdentifier().components(separatedBy: ".")[0].hasSuffix(forPolicy.lowercased())) {
-                return user
+        for account in withAccounts {
+            if (account.homeAccountId != nil && account.homeAccountId!.objectId!.hasSuffix(forPolicy.lowercased())) {
+                return account
             }
-    }
+        }
         return nil
-   }
-
-
+    }
+    
+    /**
+     
+     The way B2C knows what actions to perform for the user of the app is through the use of `Authority URL`,
+     for user in withUsers {         a URL indicating a directory that MSAL can use to obtain tokens. In Azure B2C
+     if (user.userIdentifier().components(separatedBy: ".")[0].hasSuffix(forPolicy.lowercased())) {         it is of the form `https://<instance/tfp/<tenant>/<policy>`, where `<instance>` is the
+     return user         directory host (e.g. https://login.microsoftonline.com), `<tenant>` is a
+     }         identifier within the directory itself (e.g. a domain associated to the
+     tenant, such as contoso.onmicrosoft.com), and `<policy>` is the policy you wish to
+     use for the current user flow.
+     */
+    func getAuthority(forPolicy: String) throws -> MSALB2CAuthority {
+        guard let authorityURL = URL(string: String(format: self.kEndpoint, self.kTenantName, forPolicy)) else {
+            throw NSError(domain: "MSALErrorDomain",
```

Comment:
nit: I'd prefer developers to not throw our errors directly. This would create confusion when we're checking their logs.

Reply:
agree, changed :)